logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -205,26 +185,27 @@ internal class ConnectionDataSource(
     ) = wrapStorageRequest {
         connectionDAO.insertConnection(connectionMapper.modelToDao(connection))
     }.flatMap {
-        // This can fail, but the connection will be there and get synced in worst case scenario in next SlowSync
-        wrapApiRequest {
-            userDetailsApi.getUserInfo(idMapper.toApiModel(connection.qualifiedToId))
-        }.flatMap { userProfileDTO ->
-            wrapStorageRequest {
-                val selfUser = getSelfUser()
-                val userEntity = publicUserMapper.fromUserApiToEntityWithConnectionStateAndUserTypeEntity(
-                    userDetailResponse = userProfileDTO,
-                    connectionState = connectionStatusMapper.toDaoModel(state = connection.status),
-                    userTypeEntity = userTypeEntityTypeMapper.fromTeamAndDomain(
-                        otherUserDomain = userProfileDTO.id.domain,
-                        selfUserTeamId = selfUser.teamId?.value,
-                        otherUserTeamId = userProfileDTO.teamId,
-                        selfUserDomain = selfUser.id.domain,
-                        isService = userProfileDTO.service != null
+        selfUserTeamId.invoke().flatMap { teamId ->
+            // This can fail, but the connection will be there and get synced in worst case scenario in next SlowSync
+            wrapApiRequest {
+                userDetailsApi.getUserInfo(idMapper.toApiModel(connection.qualifiedToId))
+            }.flatMap { userProfileDTO ->
+                wrapStorageRequest {
+                    val userEntity = publicUserMapper.fromUserApiToEntityWithConnectionStateAndUserTypeEntity(
+                        userDetailResponse = userProfileDTO,
+                        connectionState = connectionStatusMapper.toDaoModel(state = connection.status),
```

Comment:
state , but we provide status 🤔

Reply:
wasn't my choice :D I chose it based on the naming in the db. but will fix it.